src/p2p/node.ts

```diff
@@ -0,0 +1,66 @@
+import LibP2p from "libp2p";
+import {TCP} from "libp2p-tcp";
+import {Mplex} from "libp2p-mplex";
+import {Bootstrap} from "libp2p-bootstrap";
+import {waterfall} from "async/waterfall";
+import {PeerInfo} from "peer-info";
+import {defaultsDeep} from "@nodeutils/defaults-deep";
+import * as FloodSub from "libp2p-floodsub";
+
+export interface LodestarNodeOpts {
+  bootstrap?: any[],
+  peerInfo: PeerInfo,
+  bootnodes?: any[]
+}
+
+export class LodestarNode extends LibP2p {
+
+  private pubsub: FloodSub;
+
+  constructor(_options: LodestarNodeOpts) {
+    const defaults = {
+      modules: {
+        transport: [TCP],
+        streamMuxer: [Mplex],
+        peerDiscovery: [Bootstrap]
+      },
+      config: {
+        peerDiscovery: {
+          bootstrap: {
+            interval: 2000,
+            enabled: true,
+            list: _options.bootstrap || []
+          }
+        }
+      }
+    };
+
+    super(defaultsDeep(_options, defaults));
+  }
+
+  static async createNode(callback) {
```

Comment:
We aren't using this, do we need it?

Reply:
The idea here was to have a singleton node. I still haven't pushed these changes remotely.